app/controllers/permanent_supporting_documents_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+class PermanentSupportingDocumentsController < DashboardController
+
+  def create
+    form = PermanentSupportingDocumentForm.new(resource_params)
+    if form.save
+      flash[:success] = 'Permanent supporting document was successfully created.'
+      render status: 201, js: 'location.reload(true)'
+    else
+      render status: 422, json: { errors: form.errors.messages }
+    end
+  end
+
+  def destroy
+    permanent_document = DIDWW::Resource::PermanentSupportingDocument.load(id: params[:id])
+    if permanent_document.destroy
+      flash[:success] = 'Permanent supporting document was successfully deleted.'
+      redirect_back fallback_location: permanent_supporting_documents_path
+    else
+      flash[:danger] = 'Failed to delete Permanent supporting document: ' + permanent_document.errors[:base].join('. ')
+      redirect_back fallback_location: permanent_supporting_documents_path
```

Comment:
fallback location should be `root_path`

there is no index route for `PermanentSupportingDocumentsController`